hr_employee_document/models/hr_employee.py

```diff
@@ -35,3 +35,18 @@ def action_get_attachment_tree_view(self):
             self.env.ref("hr_employee_document.ir_attachment_view_search").id,
         )
         return action
+
+
+class HrEmployee(models.Model):
+    _inherit = "hr.employee"
+
+    @api.model
+    def check_access_rights(self, operation, raise_exception=True):
+        """We try to override access_hr_employee_system_user access right
+        only to read operation because rule_hr_employee_user rule allow access
+        only own employees"""
+        if operation == "read" and not self.env.user.has_group("hr.group_hr_user"):
```

Comment:
Why are you still giving access to employee? Permissions should be on documents, not on employee.

Reply:
Odoo allows any user to access `hr.employee.public` but not `hr.employee`, and attachments are linked to `hr.employee`